src/net.cpp

```diff
@@ -1945,9 +1919,45 @@ void static Discover(boost::thread_group& threadGroup)
 #endif
 }
 
-void StartNode(boost::thread_group& threadGroup, CScheduler& scheduler)
+CConnman::CConnman()
+{
+    setBannedIsDirty = false;
+    fAddressesInitialized = false;
+    nLastNodeId = 0;
+    nSendBufferMaxSize = 0;
+    nReceiveFloodSize = 0;
+    semOutbound = NULL;
+    nMaxConnections = 0;
+    nMaxOutbound = 0;
+    nBestHeight = 0;
+    clientInterface = NULL;
+}
+
+NodeId CConnman::GetNewNodeId()
+{
+    return nLastNodeId.fetch_add(1, std::memory_order_relaxed);
+}
+
+bool CConnman::Start(boost::thread_group& threadGroup, CScheduler& scheduler, std::string& strNodeError, Options connOptions)
```

Comment:
I'd be more comfortable with a builder doing those binds or something -- it just seems like it would be easy for a future change to introduce an unchecked dependency on the CConnman state, that could make for subtle and unpredictable bugs. But this is a complex change you're certainly more familiar with than I, so ACK however you decide to ensure safety against that kind of case.